tests/subsys/audio_modules/audio_module_template/src/template_test.c

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/fff.h>
+#include <zephyr/ztest.h>
+#include <errno.h>
+
+#include "audio_module.h"
+#include "audio_module_template.h"
+
+/* Define a timeout to prevent system locking */
+#define TEST_TX_RX_TIMEOUT_US (K_FOREVER)
+
+#define CONFIG_TEXT   "Template config"
+#define RECONFIG_TEXT "Template reconfig"
+
+#define TEST_MSG_QUEUE_SIZE	   (4)
+#define TEST_MOD_THREAD_STACK_SIZE (2048)
+#define TEST_MOD_THREAD_PRIORITY   (4)
+#define TEST_CONNECTIONS_NUM	   (5)
+#define TEST_MODULES_NUM	   (TEST_CONNECTIONS_NUM - 1)
+#define TEST_MOD_DATA_SIZE	   (40)
+#define TEST_MSG_SIZE		   (sizeof(struct audio_module_message))
+#define TEST_AUDIO_DATA_ITEMS_NUM  (20)
+
+struct mod_config {
+	int test_int1;
+	int test_int2;
+	int test_int3;
+	int test_int4;
+};
+
+struct mod_context {
+	const char *test_string;
+	uint32_t test_uint32;
+
+	struct mod_config config;
+};
+
+const char *test_instance_name = "Test instance";
+struct audio_metadata test_metadata = {.data_coding = LC3,
+				       .data_len_us = 10000,
+				       .sample_rate_hz = 48000,
+				       .bits_per_sample = 16,
+				       .carried_bits_per_sample = 16,
+				       .locations = 0x00000003,
+				       .reference_ts_us = 0,
+				       .data_rx_ts_us = 0,
+				       .bad_data = false};
+
+K_THREAD_STACK_ARRAY_DEFINE(mod_temp_stack, TEST_MODULES_NUM, TEST_MOD_THREAD_STACK_SIZE);
+DATA_FIFO_DEFINE(msg_fifo_tx, TEST_MSG_QUEUE_SIZE, TEST_MSG_SIZE);
+DATA_FIFO_DEFINE(msg_fifo_rx, TEST_MSG_QUEUE_SIZE, TEST_MSG_SIZE);
+DATA_FIFO_DEFINE(msg_fifo_tx1, TEST_MSG_QUEUE_SIZE, TEST_MSG_SIZE);
+DATA_FIFO_DEFINE(msg_fifo_rx1, TEST_MSG_QUEUE_SIZE, TEST_MSG_SIZE);
+DATA_FIFO_DEFINE(msg_fifo_tx2, TEST_MSG_QUEUE_SIZE, TEST_MSG_SIZE);
+DATA_FIFO_DEFINE(msg_fifo_rx2, TEST_MSG_QUEUE_SIZE, TEST_MSG_SIZE);
+DATA_FIFO_DEFINE(msg_fifo_tx3, TEST_MSG_QUEUE_SIZE, TEST_MSG_SIZE);
+DATA_FIFO_DEFINE(msg_fifo_rx3, TEST_MSG_QUEUE_SIZE, TEST_MSG_SIZE);
+K_MEM_SLAB_DEFINE(mod_data_slab, TEST_MOD_DATA_SIZE, TEST_MSG_QUEUE_SIZE, 4);
+
+struct data_fifo *msg_fifo_tx_array[TEST_MODULES_NUM] = {&msg_fifo_tx, &msg_fifo_tx1, &msg_fifo_tx2,
+							 &msg_fifo_tx3};
+struct data_fifo *msg_fifo_rx_array[TEST_MODULES_NUM] = {&msg_fifo_rx, &msg_fifo_rx1, &msg_fifo_rx2,
+							 &msg_fifo_rx3};
+
+ZTEST(suite_audio_module_template, test_module_template)
+{
+	int ret;
+	int i;
+	char *base_name, instance_name[CONFIG_AUDIO_MODULE_NAME_SIZE];
+
+	struct audio_data audio_data_tx;
+	struct audio_data audio_data_rx;
+
+	struct audio_module_parameters mod_parameters;
+
+	struct audio_module_template_configuration configuration = {
+		.sample_rate_hz = 48000, .bit_depth = 12345, .some_text = CONFIG_TEXT};
```

Comment:
This is just testing, but should we make the `.bit_depth` parameter a `uint8_t`? Even that is an insanely high value.

Reply:
The value is now within UINT8_T_MAX, but is the struct itself changed?

Reply:
Yes line 33 of `nrf/include/audio/audio_module_template.h`
```
struct audio_module_template_configuration {
	/* The sample rate, for testing. */
	uint32_t sample_rate_hz;

	/* The bit depth, for testing. */
	uint8_t bit_depth;

	/* A string for testing.*/
	char *module_description;
};
```